src/edu/cmu/cs/sasylf/ast/CompUnit.java

```diff
@@ -65,11 +74,13 @@ public boolean isAbstract() {
 		return !params.isEmpty();
 	}
 
-	private PackageDeclaration packageDecl;
-	private String moduleName;
-	private List<Part> params = new ArrayList<Part>();
-	private List<Part> parts = new ArrayList<Part>();
-	private int parseReports;
+	public List<Part> getParams() {

```

Comment:
I agree that returning a mutable list is suspicious. `getParams()` and `getParts()` are required because in `ModulePart`, we need to access the parameters and body of a functor when it is applied.

We should discuss how we can work around this.

Reply:
`getParams()` and `getParts()` are no longer used anywhere, which is good for abstraction.